src/ensembl/src/content/app/browser/browser-region-editor/BrowserRegionEditor.tsx

```diff
@@ -163,30 +155,36 @@ export const BrowserRegionEditor = (props: BrowserRegionEditorProps) => {
     props.replace(urlFor.browser(params));
   };
 
-  const updateBrowser = () => {
-    if (!isFormSubmitted) {
-      return;
-    }
+  const resetForm = () => {
+    updateErrorMessages(null, null);
 
+    props.toggleRegionEditorActive(false);
+    props.resetRegionValidation();
+  };
+
+  const updateBrowser = () => {
     const newChrLocation: ChrLocation = [
-      regionInput,
+      chrInput,
       getNumberWithoutCommas(locationStartInput),
       getNumberWithoutCommas(locationEndInput)
     ];
 
-    if (regionInput === region) {
+    if (chrInput === region) {
       changeLocation(newChrLocation);
     } else {
       changeFocusObject(newChrLocation);
     }
   };
 
+  const closeForm = () => {
+    updateAllInputs();
+    resetForm();
+  };
+
   const handleSubmit = (event: FormEvent<HTMLFormElement>) => {
     event.preventDefault();
 
-    const region = `${regionInput}:${locationStartInput}-${locationEndInput}`;
-
-    setIsFormSubmitted(true);
+    const region = `${chrInput}:${locationStartInput}-${locationEndInput}`;
```

Comment:
Don’t we use browserHelper for this (`getChrLocationStr`)?

Reply:
Using it would complicate the code a bit as it expects a tuple: `[string, number, number]`. However, all three of these values are strings, so I didn't find the need to use a function that returns a string.